collector/Makefile

```diff
@@ -29,28 +29,31 @@ $(GOMODULES):
 .PHONY: for-all-target
 for-all-target: $(GOMODULES)
 
+.PHONY: clean
 clean:
 	rm -rf build/
 
+.PHONY: build
 build: clean
 	@echo Building otel collector extension
 	mkdir -p $(BUILD_SPACE)/extensions
 	GOOS=linux $(GOBUILD) $(LDFLAGS) -o $(BUILD_SPACE)/extensions .
 
+.PHONY: package
 package: build
 	@echo Package zip file for collector extension layer
 	mkdir -p $(BUILD_SPACE)/collector-config
 	cp config* $(BUILD_SPACE)/collector-config
-	cd $(BUILD_SPACE) && zip -r collector-extension.zip collector-config extensions
+	cd $(BUILD_SPACE) && zip -r collector-extension-$(GOARCH).zip collector-config extensions
 
-publish-layer: package
+.PHONY: publish
+publish:
+	aws lambda publish-layer-version --layer-name $(LAYER_NAME) --zip-file fileb://$(BUILD_SPACE)/collector-extension-$(GOARCH).zip --compatible-runtimes nodejs14.x nodejs16.x nodejs18.x java11 python3.8 python3.9 --query 'LayerVersionArn' --output text
```

Comment:
Should this include the custom runtime?

Reply:
custom runtime? you mean in the list of compatible-runtimes?

Reply:
Right, to the list of runtimes.

Reply:
Shouldn't it be `publish-layer: package publish` instead of duplicating the `aws lambda publish-layer-version` command?

Reply:
According to the [aws docs](https://docs.aws.amazon.com/cli/latest/reference/lambda/publish-layer-version.html), this is the list of valid values, not seeing custom in there:

```
  nodejs
  nodejs4.3
  nodejs6.10
  nodejs8.10
  nodejs10.x
  nodejs12.x
  nodejs14.x
  nodejs16.x
  java8
  java8.al2
  java11
  python2.7
  python3.6
  python3.7
  python3.8
  python3.9
  dotnetcore1.0
  dotnetcore2.0
  dotnetcore2.1
  dotnetcore3.1
  dotnet6
  nodejs4.3-edge
  go1.x
  ruby2.5
  ruby2.7
  provided
  provided.al2
  nodejs18.x
```

Is it supported but not documented?

Reply:
I think `provided` and `provided.al2` are the custom runtimes. No idea why they are named provided or what that means :)